### PR TITLE
feat: [CDS-35933]: add serverless lambda deployment support

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
##### Summary:
- Add Serverless Lambda deployment type to select under service tab
- Support manifests and artifacts based on Serverless Lambda deployment type 
- Support AWS infra type
- Support Serverless Aws Lambda Deploy and Serverless Aws Lambda Rollback steps
- Support Basic and Blank Canvas execution strategies

##### Jira Links:
https://harness.atlassian.net/browse/CDS-35933
https://harness.atlassian.net/browse/CDS-36539
https://harness.atlassian.net/browse/CDS-34610
https://harness.atlassian.net/browse/CDS-34621
https://harness.atlassian.net/browse/CDS-34707

##### Screenshots:
Screenshot 2022-04-25 at 4 24 34 PM

Screenshot 2022-04-25 at 4 24 46 PM

Screenshot 2022-04-25 at 4 25 02 PM

Screenshot 2022-04-25 at 4 25 20 PM

Screenshot 2022-04-25 at 4 25 55 PM

Screenshot 2022-04-25 at 4 26 08 PM

Screenshot 2022-04-25 at 4 26 17 PM

Screenshot 2022-04-25 at 4 26 45 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by chetanvekariya-harness** [comment]
dockerize

**Comment by chetanvekariya-harness** [comment]
retrigger titlecheck

**Comment by chetanvekariya-harness** [comment]
trigger cypress

**Comment by chetanvekariya-harness** [comment]
dockerize

**Comment by chetanvekariya-harness** [comment]
dockerize

**Comment by chetanvekariya-harness** [comment]
dockerize

**Comment by chetanvekariya-harness** [comment]
dockerize

**Comment by chetanvekariya-harness** [comment]
trigger cypress

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts [hunk 1]
```diff
@@ -75,9 +77,10 @@ export interface ImagePathTypes {
   registryHostname?: string
   region?: any
   repositoryPort?: number | string
-  repository?: string
+  repository?: any
```
Comment (diptiman-adak) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts:80]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
  repository?: unknown
```

File: src/modules/75-cd/components/PipelineSteps/ServerlessInfraSpec/ServerlessInfraSpec.tsx [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+  const { getString } = useStrings()
+
+  const { subscribeForm, unSubscribeForm } = React.useContext(StageErrorContext)
+
+  const formikRef = React.useRef<FormikProps<unknown> | null>(null)
+
+  React.useEffect(() => {
+    subscribeForm({ tab: DeployTabs.INFRASTRUCTURE, form: formikRef })
+    return () => unSubscribeForm({ tab: DeployTabs.INFRASTRUCTURE, form: formikRef })
+  }, [])
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/ServerlessInfraSpec/ServerlessInfraSpec.tsx:88]: Codacy has a fix for the issue: [React Hook React.useEffect has missing dependencies: 'subscribeForm' and 'unSubscribeForm'. Either include them or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
  }, [subscribeForm, unSubscribeForm])
```

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts [hunk 3]
```diff
@@ -129,46 +130,93 @@ export const getFinalArtifactObj = (
...
     }
   }
+
   if (isSideCar) {
     merge(artifactObj, { identifier: formData?.identifier })
   }
   return artifactObj
 }
 
+const getTagValues = (specValues: any, isServerlessDeploymentTypeSelected = false): ImagePathTypes => {
```
Comment (diptiman-adak) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts:180]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
const getTagValues = (specValues: unknown, isServerlessDeploymentTypeSelected = false): ImagePathTypes => {
```

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/K8sServiceSpecForms/KubernetesServiceSpecEditable.tsx [hunk 4]
```diff
@@ -20,6 +21,25 @@ import { setupMode } from '../K8sServiceSpecHelper'
 import type { KubernetesServiceInputFormProps } from '../K8sServiceSpecInterface'
 import css from '../K8sServiceSpec.module.scss'
 
+const getManifestCardHeader = (
+  selectedDeploymentType: string,
+  getString: (key: keyof StringsMap, vars?: Record<string, any> | undefined) => string
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/K8sServiceSpecForms/KubernetesServiceSpecEditable.tsx:26]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
  getString: (key: keyof StringsMap, vars?: Record<string, unknown> | undefined) => string
```

**Comment by chetanvekariya-harness** [comment]
dockerize

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/ServerlessArtifactoryRepository.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  }, [artifactRepoLoading, artifactRepoError])
+
+  useEffect(() => {
+    if (getMultiTypeFromValue(connectorRef) === MultiTypeInputType.FIXED && !artifactRepoData) {
+      getArtifactRepos()
+    }
+    if (artifactRepoData) {
+      setConnectorRepos(map(artifactRepoData.data?.repositories, repo => ({ label: repo, value: repo })))
+    }
+  }, [artifactRepoData, connectorRef])
```
Comment (diptiman-adak) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/ServerlessArtifactoryRepository.tsx:91]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'getArtifactRepos'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
  }, [artifactRepoData, connectorRef, getArtifactRepos])
```

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx [hunk 6]
```diff
@@ -32,7 +44,107 @@ import ArtifactTagRuntimeField from '../ArtifactSourceRuntimeFields/ArtifactTagR
...
+  fromTrigger?: boolean
+  artifact?: PrimaryArtifact | SidecarArtifact
+  selectedDeploymentType: ServiceDeploymentType
+  isSidecar?: boolean
+  artifactPath?: string
+  isTagsSelectionDisabled: (data: ArtifactSourceRenderProps, isServerlessDeploymentTypeSelected: boolean) => boolean
+  fetchingTags: boolean
+  artifactoryTagsData: ResponseArtifactoryResponseDTO | null
+  fetchTagsError: GetDataError<Failure | Error> | null
+  fetchTags: (props?: any) => Promise<void> | undefined
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx:64]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
  fetchTags: (props?: unknown) => Promise<void> | undefined
```

**Comment by chetanvekariya-harness** [comment]
dockerize

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx (hunk 6 above)
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx:64]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
  fetchTags: (props?: unknown) => Promise<void> | undefined
```

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx [hunk 7]
```diff
@@ -69,27 +75,88 @@ const CardList = ({ items, isReadonly, selectedValue, onChange, allowDisabledIte
...
+    cgTypes = cgTypes.filter(deploymentType => deploymentType.value !== 'NativeHelm')
+  }
+  if (serverlessFF) {
+    cgTypes = cgTypes.filter(deploymentType => !isServerlessDeploymentType(deploymentType.value))
+  }
+  return cgTypes
+}
+
+const getServerlessDeploymentTypes = (
+  getString: (key: keyof StringsMap, vars?: Record<string, any> | undefined) => string,
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType.tsx:106]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
  getString: (key: keyof StringsMap, vars?: Record<string, unknown> | undefined) => string,
```

**Comment by PragyeshMishra01** [comment]
dockerize

**Comment by PragyeshMishra01** [comment]
dockerize

**Comment by PragyeshMishra01** [comment]
dockerize

**Comment by chetanvekariya-harness** [comment]
dockerize

**Comment by chetanvekariya-harness** [comment]
trigger cypress

**Review by bhavya-sinha — APPROVED** [review]

**Comment by chetanvekariya-harness** [comment]
trigger cypress

**Comment by chetanvekariya-harness** [comment]
dockerize

**Comment by chetanvekariya-harness** [comment]
dockerize

**Review by bhavya-sinha — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**